File: Corrections/php/Neufplate/src/AvatarProvider/DiceBear/DiceBearClient.php
<?php

namespace App\AvatarProvider\DiceBear;

use App\AvatarClientInterface;
use App\AvatarProvider\SpriteType;

class DiceBearClient implements AvatarClientInterface
{
    private static ?DiceBearClient $_instance = null;

    public string $spriteType;
    private const apiUrl = "https://avatars.dicebear.com/api/";
    private const fileType = ".svg";

    public function __construct()
    {
        $this->spriteType = SpriteType::HUMAN->value;
    }

    public static function getInstance(): AvatarClientInterface {

        if(is_null(self::$_instance)) {
            self::$_instance = new DiceBearClient();
        }

        return self::$_instance;
    }

    // Exercice 1
    public static function randomAvatarUrl(): string
    {
        return self::apiUrl . self::getInstance()->spriteType . "/" . uniqid() . self::fileType . PHP_EOL;
    }

    // Exercice 2
    public function getRandomAvatarUrl(): string
    {
        return self::apiUrl . self::getInstance()->spriteType . "/" . uniqid() . self::fileType . PHP_EOL;
    }

    public function setSpriteType(SpriteType $spriteType): void
    {
        $this->spriteType = $spriteType->value;
    }

    public function getAvatarFromHash(string $hash): string
    {
        return self::apiUrl . "/" . $hash . self::fileType . PHP_EOL;
    }

}

File: Corrections/java/Neufplate/src/main/java/fr/neufplate/AvatarProvider/DiceBear/DiceBearClient.java
package fr.neufplate.AvatarProvider.DiceBear;

import fr.neufplate.AvatarClientInterface;
import fr.neufplate.AvatarProvider.SpriteType;

public final class DiceBearClient implements AvatarClientInterface {

    private static DiceBearClient instance;

    public String spriteType = SpriteType.HUMAN.value;
    private static final String apiUrl = "https://avatars.dicebear.com/api/";
    private static final String fileType = ".svg";

    public static DiceBearClient getInstance() {
        if (instance == null) {
            instance = new DiceBearClient();
        }
        return instance;
    }

    public void setSpriteType(SpriteType spriteType) {
        this.spriteType = spriteType.value;
    }


    // Exercice 1
    public static String randomAvatarUrl() {
        return apiUrl + getInstance().spriteType + "/" + Math.random() + fileType;
    }

    // Exercice 2
    public String getRandomAvatarUrl() {
        return apiUrl + getInstance().spriteType + "/" + Math.random() + fileType;
    }

    @Override
    public String getAvatarFromHash(String hash) {
        return apiUrl + getInstance().spriteType + "/" + hash + fileType;
    }


}